src/bindings/python/src/openvino/runtime/opset13/ops.py

```diff
@@ -222,3 +229,53 @@ def scaled_dot_product_attention(
         "causal": causal,
     }
     return _get_node_factory_opset13().create("ScaledDotProductAttention", inputs, attributes)
+
+
+@nameable_op
+def constant(
+    value: Union[NumericData, np.number, bool, np.bool_, list],
+    dtype: Union[NumericType, Type] = None,
+    name: Optional[str] = None,
+    *,
+    shared_memory: bool = False,
+) -> Constant:
+    """Create a Constant node from provided value.
+
+    :param value: One of: array of values or scalar to initialize node with.
+                  Note: if value is a scalar, shared memory is disabled.
+    :param dtype: The data type of provided data.
+                  If dtype does not match, data will be converted.
+                  Note: disables sharing of the memory when convertion occurs.
+    :param name: Optional name for output node.
+    :param shared_memory: keyword-only argument.
+                          If `True`, this Constant's memory is being shared with a host,
+                          that means the responsibility of keeping host memory is
+                          on the side of a user. Any action performed on the host
+                          memory is reflected on this Constant's memory!
+                          If `False`, data is being copied to this Constant.
+                          Requires data to be C_CONTIGUOUS if `True`.
+    :return: The Constant node initialized with provided data.
+    """
+    if isinstance(value, np.ndarray):
+        _value, _shared_memory = value, shared_memory
+    else:
+        _value, _shared_memory = np.array(value), False
+        log.warning(f"Converting scalar to corresponding type of {_value.dtype}. Memory sharing is disabled by default.")
+    # Handle type casting, when dtype is not None:
+    if dtype:
+        _dtype = dtype.to_dtype() if isinstance(dtype, Type) else dtype
+        if _dtype is int:
+            log.warning("Converting scalar type of undefined bitwidth to 32-bit integer. Memory sharing is disabled by default.")
+            _value, _shared_memory = _value.astype(np.int32), False
+        elif _dtype is float:
+            log.warning("Converting scalar type of undefined bitwidth to 32-bit float. Memory sharing is disabled by default.")
+            _value, _shared_memory = _value.astype(np.float32), False
+        elif _dtype is bool:
+            log.warning("Converting bool type to numpy bool. Memory sharing is disabled by default.")
+            _value, _shared_memory = _value.astype(np.bool_), False
+        else:
+            if _dtype != _value.dtype:
+                log.warning(f"Converting value of {_value.dtype} to {_dtype}. Memory sharing is disabled by default.")
+                _value, _shared_memory = _value.astype(_dtype), False
+    # Create Constant itself:
+    return Constant(_value, shared_memory=_shared_memory)
```

Comment:
Done

Reply:
The element type is still not passed to the Constant constructor for the regular types, so what is going to happen if the `dtype` of the numpy array doesn't match the type provided as `dtype` attr of the py constant op?
For example:
`opset13.constant(np.array([1]), dtype=ov.Type.i32)` (np.int64 for numpy array by default)
`opset13.constant(np.array([9223372036854775807], dtype=np.int64), dtype=ov.Type.i32)`.

Seems like the output element type of the returned ov Constant will be silently concluded from the numpy array (in the example as `ov.Type.i64`), ignoring the requested `dtype`.

Reply:
Discussed offline, behavior has changed according to underlaying C++ constructors. It was throwing an error because of specialized casting constructors which called `fill_data` which is not present in newer shared buffers and void* constructors.